utils/scheduler.py

```diff
@@ -0,0 +1,29 @@
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+from utils.core import get_random_quote
+import logging
+from logging.config import fileConfig
+
+fileConfig('../utils/logging_config.ini')
+logger = logging.getLogger('discord')
+
+
+def init_message_scheduler(config, client):
+
+    jobs = config.get("SCHEDULES")
+    if not jobs or not jobs[0]["MESSAGES"]:
```

Comment:
This check will throw ```IndexError``` if you have SCHEDULES key, but no defined triggers. Consider the config: 

```python
{
    "SCHEDULES": [ ]
}
```
This will get the jobs item, but has an empty list. 

It is also assuming that it has correctly defined "MESSAGES" as a key in the first item, and will return a ```KeyError``` if it is not.

I recommend two possible actions:

1. Instead of checking the truthiness of the first object, just check if there are elements in the list with len(jobs). This tells you that you aren't being given an empty list, which I anticipate will be common.  

Note: This removes the check for MESSAGES as a key in the config. However, I believe this is only partially helpful anyway, as it will only verify the first message is correct. We can add a check to be sure all jobs are correctly formatted with something like:

```python
if not jobs or not len(jobs) or not all("MESSAGES" in job.keys() for job in jobs)
```
If the intention is to move forward is *any* of the jobs are good, then we can adjust that slightly to just be:

```python
if not jobs or not len(jobs) or not any("MESSAGES" in job.keys() for job in jobs):
```
I am not sure which of those I prefer, but I will leave that up to you to consider in the broader context of the purpose of this check.

2. We can always choose to *not* perform this check, as it may be harmless to move forward with scheduling jobs, since they will never be added if this list is empty. The only side effect of this is that you will still start the scheduler, but this seems harmless so long as no unwanted jobs are scheduled. There is the added caveat that we must be careful to check we have a correctly formatted job later if we do not check it here.